### PR TITLE
[chore] Add backward-compatibility check to ci

### LINKED ISSUE #740: Formalize attribute deprecation, restrict removal
One of the key takeaways from #551 and similar issues on the tooling is that removing an attribute from semconv results in breaking changes for all languages.

Codegen (jinja based) does not take into account previous versions of the spec and does not generate code for removed attributes.

Given that semconv artifact is used by user applications and by different instrumentation libraries (potentially of different versions), removing an attribute leads to breaking changes that are too annoying even for unstable artifact.

The workaround SIGs implemented is to (manually) preserve removed attributes and annotate them as deprecated which created a lot of  unnecessary work (and pain).

**Proposal**
- Do not remove attributes, deprecate them
- `Stable` -> `deprecated` attributes should stay till the next major semconv version (forever 🤞)
- We can decide to remove `experimental` -> `deprecated` attributes after X versions/months
- Add automated checks to semconv repo to prevent attribute removal

Nice-to-haves:
- add yaml annotations like `deprecated_since` (and maybe `to_be_removed_in` )

### PR DESCRIPTION
**Pull request by lmolkova**
Fix #740

## Changes

Adds basic back-compat for semconv against previous version check to CI.

Currently, it's just a demonstration of https://github.com/open-telemetry/build-tools/pull/271 which is under review.

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] Change log entry added, according to the guidelines in [When to add a changelog entry](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#when-to-add-a-changelog-entry).
  * If your PR does not need a change log, start the PR title with `[chore]`
* ~~[schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.~~

### CODE REVIEW

**Review by jsuereth — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -103,6 +105,11 @@ table-check:
 	docker run --rm -v $(PWD)/model:/source -v $(PWD)/docs:/spec \
 		otel/semconvgen:$(SEMCONVGEN_VERSION) -f /source markdown -md /spec --md-check --md-use-badges --md-stable
 
+.PHONY: compatibility-check
+compatibility-check:
```
Comment (lmolkova) [inline Makefile:109]: TODO:

based on https://github.com/open-telemetry/semantic-conventions/pull/770#discussion_r1502392874, I will also update contributing guide to add instructions on how to run it and how to deprecate attributes/metrics

**Review by trisch-me — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -103,6 +105,11 @@ table-check:
 	docker run --rm -v $(PWD)/model:/source -v $(PWD)/docs:/spec \
 		otel/semconvgen:$(SEMCONVGEN_VERSION) -f /source markdown -md /spec --md-check
 
+.PHONY: compatibility-check
+compatibility-check:
+	docker run --rm -v $(PWD)/model:/source -v $(PWD)/docs:/spec \
+		otel/semconvgen:$(SEMCONVGEN_VERSION) -f /source compatibility --previous-version ${LASTEST_RELEASED_SEMCONV_VERSION} --ignore-warnings
```
Comment (trask) [inline Makefile:111]:
```suggestion
		otel/semconvgen:$(SEMCONVGEN_VERSION) -f /source compatibility --previous-version ${LATEST_RELEASED_SEMCONV_VERSION} --ignore-warnings
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -300,6 +301,18 @@ To check the validity of links in all markdown files, run the following command:
 make markdown-link-check
 ```
 
+### Version compatibility check
+
+Semantic conventions are validated for backward compatibility with last released versions. Here's [the full list of compatibility checks](https://github.com/open-telemetry/build-tools/blob/main/semantic-conventions/README.md#version-compatibility-check).
+Removing attributes, metrics, or enum members is not allowed, they should be deprecated instead.
+It applies to stable and experimental conventions and prevents semantic conventions auto-generated libraries from introducing breaking changes.
+
+You can run backward compatibility check in all yaml files with the following command:
```
Comment (trask) [inline CONTRIBUTING.md:310]:
```suggestion
You can run the backward compatibility check in all yaml files with the following command:
```

**Review by trask — APPROVED** [review]
💯

File: Makefile [hunk 4]
```diff
@@ -15,9 +15,11 @@ CHLOGGEN_CONFIG  := .chloggen/config.yaml
 # Keep links in model/README.md and .vscode/settings.json in sync!
 SEMCONVGEN_VERSION=0.24.0
 
+LASTEST_RELEASED_SEMCONV_VERSION=1.25.0
```
Comment (lmolkova) [inline Makefile:18]:
```suggestion
LATEST_RELEASED_SEMCONV_VERSION=1.25.0
```

File: Makefile [hunk 5]
```diff
@@ -17,7 +17,7 @@ SEMCONVGEN_VERSION=0.24.0
 
 # TODO: add `yamllint` step to `all` after making sure it works on Mac.
 .PHONY: all
-all: install-tools markdownlint markdown-link-check misspell table-check schema-check \
+all: install-tools markdownlint markdown-link-check misspell table-check compatibility-check schema-check \
```
Comment (joaopgrassi) [inline Makefile:20]: Nit: Maybe also add to `make check`

**Review by joaopgrassi — APPROVED** [review]

**Review by pyohannes — APPROVED** [review]

**Pull request merged**